pkg/updater/service/datadog_agent.go

```diff
@@ -120,3 +137,78 @@ func StartAgentExperiment() error {
 func StopAgentExperiment() error {
 	return startUnit(agentUnit)
 }
+
+func setInstallInfo() {
+	// avoid rewriting the files if they already exist
+	if _, err := os.Stat(installInfoFile); err == nil {
+		log.Info("Install info file already exists, skipping")
+		return
+	}
+	tool, version := getToolVersion()
+	if err := writeInstallInfo(tool, version); err != nil {
+		log.Warnf("Failed to write install info: %s", err)
+	}
+	if err := writeInstallSignature(); err != nil {
+		log.Warnf("Failed to write install signature: %s", err)
+	}
+}
+
+func cleanInstallInfo() {
+	if err := os.Remove(installInfoFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install info file: %s", err)
+	}
+	if err := os.Remove(installSigFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install signature file: %s", err)
+	}
+}
+
+func getToolVersion() (string, string) {
+	tool := "unknown"
+	version := "unknown"
+	if _, err := exec.LookPath("dpkg-query"); err == nil {
+		tool = "dpkg"
+		toolVersion, err := getDpkgVersion()
+		if err == nil {
+			version = toolVersion
+		}
+	}
+	return tool, version
+}
+
+func getDpkgVersion() (string, error) {
+	cmd := exec.Command("dpkg-query", "--showformat=${Version}", "--show", "dpkg")
+	output, err := cmd.Output()
+	if err != nil {
+		log.Warnf("Failed to get dpkg version: %s", err)
+		return "", err
+	}
+	splitVersion := strings.Split(strings.TrimSpace(string(output)), ".")
+	if len(splitVersion) < 3 {
+		return "", fmt.Errorf("failed to parse dpkg version: %s", string(output))
+	}
+	return strings.Join(splitVersion[:3], "."), nil
+}
+
+func writeInstallInfo(tool, version string) error {
+	installInfoContent := fmt.Sprintf(`---
+install_method:
+  tool: %s
+  tool_version: %s
+  installer_version: updater_package
+  installer: updater
+`, tool, version)
+	return os.WriteFile(installInfoFile, []byte(installInfoContent), 0644)
+}
+
+func writeInstallSignature() error {
+	installSignature := map[string]string{
+		"install_id":   strings.ToLower(uuid.New().String()),
+		"install_type": "manual_update_via_apt",
```

Comment:
this should be updated later to the different flavors
- updater through install script
- manual bootstrap call
- updater through manual apt/rpm